ide/nativeimage.api/src/org/netbeans/modules/nativeimage/api/debug/NIDebuggerService.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.nativeimage.api.debug;
+
+import org.netbeans.api.annotations.common.CheckForNull;
+import org.netbeans.api.project.Project;
+import org.netbeans.modules.nativeimage.api.NIUtils;
+import org.netbeans.modules.nativeimage.spi.debug.NIDebuggerProvider;
+import org.netbeans.modules.nativeimage.spi.debug.NIDebuggerServiceProvider;
+import org.openide.filesystems.FileObject;
+import org.openide.util.Lookup;
+
+/**
+ * A service that provides native image debugger.
+ */
+public final class NIDebuggerService {
+
+    private final NIDebuggerServiceProvider provider;
+
+    private NIDebuggerService(NIDebuggerServiceProvider provider) {
+        this.provider = provider;
+    }
+
+    /**
+     * Checks whether native image debugging is available.
+     * @return <code>true</code> when there is an implementation of
+     *         {@link NIDebuggerServiceProvider} registered in the default lookup.
+     */
+    public static boolean isAvailable() {
```

Comment:
A bit useless: `getDefault() != null` is likely serve the same purpose. On _slow_ as well as _fast path_.

Reply:
Removed.